### PR TITLE
#29 rework/Booking Store

### PR DESCRIPTION
**Pull request by mathvaillant**
## Summary of the changes made:

- [x] Zustand setup tests.
- [x] Tests for the `useBookingStore`. 
- [x] Refactored the `useBookingStore` to handle ticket group types, to be simpler and to hold the full `route` and `departure` data, not only pieces of those.  
- [x] Updated some types to be more context meaningful. 
- [x] Fixed: Refreshing the app goes to Route Selection and shows no options

### CODE REVIEW

File: src/hooks/useBookingStore.tsx [hunk 1]
```diff
@@ -1,58 +1,98 @@
...
+  route?: RouteResponse;
+  departure?: DepartureResponse;
+  setRoute: (route: RouteResponse) => void;
+  setDeparture: (departure: DepartureResponse) => void;
+  itemCounters: ItemCounters;
+  resetCounters: () => void;
+  increment: (group: TicketTypeGroup["name"], ticket: TicketToSell) => void;
+  decrement: (group: TicketTypeGroup["name"], ticket: TicketToSell) => void;
+  getTickets: () => Ticket[];
+  getGroupTickets: () => {
```
Comment (mathvaillant) [inline src/hooks/useBookingStore.tsx:27]: This `getGroupTickets` will be necessary in order to display the data on this component when it needs to render the tickets: 
image

File: src/hooks/useBookingStore.tsx [hunk 2]
```diff
@@ -1,58 +1,98 @@
...
-  decrementItemCountersKey: (key: string) => void;
+  route?: RouteResponse;
+  departure?: DepartureResponse;
+  setRoute: (route: RouteResponse) => void;
+  setDeparture: (departure: DepartureResponse) => void;
+  itemCounters: ItemCounters;
+  resetCounters: () => void;
+  increment: (group: TicketTypeGroup["name"], ticket: TicketToSell) => void;
+  decrement: (group: TicketTypeGroup["name"], ticket: TicketToSell) => void;
+  getTickets: () => Ticket[];
```
Comment (mathvaillant) [inline src/hooks/useBookingStore.tsx:26]: This `getTickets` is necessary so we don't have to do any logic on the Payment screen, only get the tickets and create the booking.

File: src/screens/Booking/Booking.tsx [hunk 3]
```diff
@@ -1,64 +1,68 @@
...
-  };
+  // TODO: This is handled in another PR
+  const onPressBook = () => navigate(NavigationScreens.BOOKING_SUMMARY);
 
   return (
     <ScreenLayout>
       <View style={styles.header}>
         <Typography size="sm" style={styles.headerText}>
-          {t("common.voyageleg")}: {departureDate} {departureTime} {originCode} - {destinationCode}
+          {/* TODO: Turn this into a custom component */}
```
Comment (mathvaillant) [inline src/screens/Booking/Booking.tsx:48]: I'll have this done on the #23

File: src/screens/__tests__/Booking.test.tsx [hunk 4]
```diff
@@ -14,31 +18,33 @@ jest.mock("@react-navigation/native", () => {
...
+
 describe("Booking Screen", () => {
   it("renders the screen with header and footer buttons", async () => {
     const { getByText, getAllByTestId } = render(<BookingScreen />);
 
     await waitFor(() => {
-      expect(
-        getByText(RegExp(`${DATE} ${TIME} ${ORIGIN_CODE} - ${DESTINATION_CODE}`)),
-      ).toBeTruthy();
+      // TODO: This will work differently when we use a custom component for the Screen header
```
Comment (mathvaillant) [inline src/screens/__tests__/Booking.test.tsx:44]: Once I have the custom component made on the #23 I'll update these tests

**Review by joaoqueiroz95 — APPROVED** [review]
LGTM

**Comment by mathvaillant** [comment]
[Revisit ItemCounters structure](https://trello.com/c/BMilnw7v/29-revisit-itemcounters-structure)

**Comment by joaoqueiroz95** [comment]
[Booking Summary Screen](https://trello.com/c/wPZtaG1x/35-booking-summary-screen)

**Pull request merged**